src/server/api.test.ts

```diff
@@ -472,6 +484,269 @@ describe('.createApiServer', () => {
+                        credentials: 'SECRET1',
+                      },
+                      '1': {
+                        name: 'bob',
+                        credentials: 'SECRET2',
+                      },
+                    },
+                  },
+                };
+              },
+            });
+            const app = createApiServer({ db, games });
+            response = await request(app.callback())
+              .post('/games/foo/1/update')
+              .send('playerID=0&credentials=SECRET1&newName=ali');
+          });
+
+          test('is successful', async () => {
+            expect(response.status).toEqual(200);
+          });
+
+          test('updates the players', async () => {
+            expect(db.mocks.setMetadata).toHaveBeenCalledWith(
+              '1',
+              expect.objectContaining({
+                players: expect.objectContaining({
+                  '0': expect.objectContaining({
+                    name: 'ali',
+                  }),
+                }),
+              })
+            );
+          });
+        });
+
+        describe('when the playerID does not exist', () => {
+          test('player not found', async () => {
+            const app = createApiServer({ db, games });
+            response = await request(app.callback())
+              .post('/games/foo/1/update')
+              .send('playerID=2&credentials=SECRET1&newName=joe');
+            expect(response.text).toEqual('Player 2 not found');
+          });
+        });
+
+        describe('when the credentials are invalid', () => {
+          test('invalid credentials', async () => {
+            const app = createApiServer({ db, games });
+            response = await request(app.callback())
+              .post('/games/foo/1/update')
+              .send('playerID=0&credentials=SECRET2&newName=mike');
+            expect(response.text).toEqual('Invalid credentials SECRET2');
+          });
+        });
+        describe('when playerID is omitted', () => {
+          beforeEach(async () => {
+            const app = createApiServer({ db, games });
+            response = await request(app.callback())
+              .post('/games/foo/1/update')
+              .send('credentials=foo&newName=bill');
+          });
+
+          test('playerID is required', async () => {
+            expect(response.text).toEqual('playerID is required');
+          });
+          describe('when newName is omitted', () => {
+            beforeEach(async () => {
+              const app = createApiServer({ db, games });
+              response = await request(app.callback())
+                .post('/games/foo/1/update')
+                .send('credentials=foo&playerID=0');
+            });
+
+            test('newName is required', async () => {
+              expect(response.text).toEqual('newName or data is required');
+            });
+          });
+        });
+      });
+    });
+  });
+
+  describe('updating', () => {
```

Comment:
Let’s name this to match the handler you wrote, e.g. “updating player metadata”.